### PR TITLE
Declare strict types

### PR DESCRIPTION
**Pull request by gmponos**
I could also add it as a php-cs rule but checking travis I saw that you don't check the code against php-cs

### CODE REVIEW

**Comment by gmponos** [comment]
Ah!.. you use pretty ci... ok.. so shall I add it? if I do I will need to enable risky rules.

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/.phpunit.result.cache
```
Comment (gmponos) [inline .gitignore]: Added this also on this PR.

**Comment by gmponos** [comment]
Done.

File: src/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -270,222 +264,219 @@ private function createClientPluginNode()
...
-                            ->end()
-                        ->end()
-                    ->end() // End naming_strategy_options
-                    ->scalarNode('fixtures_directory')
-                        ->info('Where the responses will be stored and replay from when using the filesystem recorder. Should be accessible to your VCS.')
-                    ->end()
-                ->end()
-            ->end()
-        ->end();
+            ->arrayNode('reference')
```
Comment (Nyholm) [inline src/DependencyInjection/Configuration.php]: Oh,, I do not like these changes. I liked the tree structure

Comment (gmponos) [reply src/DependencyInjection/Configuration.php]: Wait.. I see more changes that I didn't do... :/

Comment (gmponos) [reply src/DependencyInjection/Configuration.php]: OK. these changes are made from php-cs-fixer... :/ 

Maybe allowing strict rules enables some rules of symfony that were not previously enabled.. :/

I would suggest not to add the rule on php-cs

**Comment by gmponos** [comment]
> What are the benefits of having declare(strict_types=1)?

I guess today none.. but why not having it?

File: src/DependencyInjection/Configuration.php (hunk 2 above)
Comment (dbu) [reply src/DependencyInjection/Configuration.php]: can you revert the whitespace problem, push and see if pretty-ci wants to destroy it again?

**Comment by Nyholm** [comment]
> but why not having it?

Because if we made one small mistake, everything will break horribly. =)

File: src/DependencyInjection/Configuration.php (hunk 2 above)
Comment (gmponos) [reply src/DependencyInjection/Configuration.php]: Pretty ci passes.. another reason that I am thinking is that ci is using older php-cs version.

would be nice if the php-cs could be included as a phar in the project for these kind of reasons.. developers will be able to contribute easier.

**Comment by gmponos** [comment]
> Because if we made one small mistake, everything will break horribly. =)

Ah.. I got to admit that you have a point and better safe than sorry.. then may I suggest in order not to get all the work lost.. to leave it for tests and interfaces? or you want to disable except if you want to have a php-cs rule for having it removed.

**Comment by dbu** [comment]
i would be for having the strict types. if we do have mistakes, people will tell us, and we can fix them. more likely, other people have mistakes and can notice and fix them. non-strictness essentially hides issues. i would see this in a new minor version, not as a patch release.

most code in the bundle is not interacted with by the user, but is about the collector and factories and such.

File: src/Collector/Collector.php [hunk 3]
```diff
@@ -1,10 +1,13 @@
 <?php
 
+declare(strict_types=1);
+
 namespace Http\HttplugBundle\Collector;
 
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\HttpKernel\DataCollector\DataCollector;
+use Symfony\Component\HttpKernel\Kernel;
```
Comment (dbu) [inline src/Collector/Collector.php]: this line should not be added, its an unused import

**Comment by dbu** [comment]
@Nyholm are you ok if we merge this? afaik it should only cause BC breaks if we have programming errors in our code, and if we do we rather know and fix them than ignoring the problem.

**Comment by Nyholm** [comment]
Fair. @gmponos, can you make sure the ci is green

**Comment by dbu** [comment]
also needs a rebase on master again, to solve conflicts :-/

**Pull request merged**